activemq-broker/src/main/java/org/apache/activemq/broker/TransactionBroker.java

```diff
@@ -291,13 +294,43 @@ public void send(ProducerBrokerExchange producerExchange, final Message message)
             transaction = getTransaction(context, message.getTransactionId(), false);
         }
         context.setTransaction(transaction);
+
         try {
+            // [AMQ-9344] Limit uncommitted transactions by count
+            verifyUncommittedCount(producerExchange, transaction, message);
             next.send(producerExchange, message);
         } finally {
             context.setTransaction(originalTx);
         }
     }
 
+    protected void verifyUncommittedCount(ProducerBrokerExchange producerExchange, Transaction transaction, Message message) throws Exception {
+        // maxUncommittedCount <= 0 disables
+        int maxUncommittedCount = this.getBrokerService().getMaxUncommittedCount();
```

Comment:
You should be able to just read this in the constructor and store it as a final variable on the TransactionBroker as it won't be changed later since there's no dynamic config for the property

Reply:
That should be fine since it's just a getter I was just figuring it's pretty unlikely it would ever be changed, it would be rare as likely you are going to set the limit to something reasonable in the first place.

Reply:
Correct. Changing the value at runtime would be 'emergency playbook' or 'testing different scenarios' for the most part.